### PR TITLE
[WIP] Support for reading PE32 ObjDumps provided there's a useful .map file

### PR DESCRIPTION
**Pull request by partouf**
Work in progress - but so far I can read Windows .exe objdumps and match the functions and lines if there's a .map file.

Currently just tested with binaries and maps generated by Delphi 7, but planning to try and support MSVC binaries and maps as well.

Delphi 7 binary with labels and line number matching:
FPC vs Delphi comparison

Todo:
- [x] Basic MAP support Windows
- [x] Basic MAP support Delphi
- [ ] Setup so that binary mode is always enabled and can't be disabled
- [ ] Make a local properties that supports all compilers
- [ ] Write examples with generics that compile on both fpc and delphi
- [ ] Delphi 5
- [ ] Delphi 6
- [x] Delphi 7
- [ ] Delphi 2007
- [x] Delphi 2009
- [x] Delphi XE
- [ ] Delphi XE2
- [ ] Delphi XE3
- [ ] Delphi XE4
- [ ] Delphi XE5
- [ ] Delphi XE6
- [ ] Delphi XE7
- [ ] Delphi XE8
- [ ] Delphi 10 Seattle
- [ ] Delphi 10.1 Berlin
- [ ] Delphi 10.2 Tokyo
- [ ] Ask terms/permission from Embarcadero to use some compilers on godbolt.org

### CODE REVIEW

File: lib/compilers/pascal-win.js [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+        //     '/FAsc',
+        //     '/c',
+        //     '/Fa' + this.filename(outputFilename),
+        //     '/Fo' + this.filename(outputFilename + '.obj')
+        // ];
+    };
+    return compile.initialise();
+}
+
+module.exports = compilePascalWin32;
```
Comment (RabsRincon) [inline lib/compilers/pascal-win.js]: Newline

File: etc/config/pascal.amazon.properties [hunk 2]
```diff
@@ -1,10 +1,9 @@
-compilers=&fpc
+compilers=&fpc:&delphi
 defaultCompiler=fpc302
 
 group.fpc.compilers=fpc260:fpc262:fpc264:fpc302
 group.fpc.options=@/opt/compiler-explorer/fpc/fpc.cfg -g -al
 group.fpc.demangler=/dev/null
-group.fpc.objdumper=/opt/compiler-explorer/gcc-7.2.0/bin/objdump
```
Comment (RabsRincon) [inline etc/config/pascal.amazon.properties]: No more objdump?

Comment (partouf) [reply etc/config/pascal.amazon.properties]: ah, i think this was just for testing to make it run on my environment, will undo

File: lib/map-file.js [hunk 3]
```diff
@@ -0,0 +1,381 @@
...
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+"use strict";
+
+var fs = require("fs");
+
+class MapFileReader {
+    /**
```
[the remark this answers is not in the document]
Comment (partouf) [reply lib/map-file.js]: Visual Studio Code

**Comment by partouf** [comment]
Progress so far:

D7vsD2009vsDXE

**Comment by RabsRincon** [comment]
Hi! As part of our plans for February, we are trying to reduce our number of long standing pull requests. We would like to know if you are planning on working on this, or if we should try to finish it for you (Or close it in case we are unable to).

We don't want to push you into doing something you might not want to do/don't have the time right now, so we'll understand if you choose not to work on this. No hard feelings :). We are indeed grateful you took the time to try to improve CE.

**Comment by partouf** [comment]
closing this until i have more time

**Pull request closed without merging**